app/assets/javascripts/index/directions.js

```diff
@@ -109,31 +119,64 @@ OSM.Directions = function (map) {
 
       var viewbox = map.getBounds().toBBoxString(); // <sw lon>,<sw lat>,<ne lon>,<ne lat>
 
-      $.getJSON(OSM.NOMINATIM_URL + "search?q=" + encodeURIComponent(endpoint.value) + "&format=json&viewbox=" + viewbox, function (json) {
-        endpoint.awaitingGeocode = false;
-        endpoint.hasGeocode = true;
-        if (json.length === 0) {
-          input.addClass("is-invalid");
-          alert(I18n.t("javascripts.directions.errors.no_place", { place: endpoint.value }));
-          return;
-        }
-
-        endpoint.setLatLng(L.latLng(json[0]));
+      if (endpoint.value.match(new RegExp("^" + coordinatesRegularExpression + "$"))) {
+        var latlng_array = endpoint.value.split(/[/,]/);
+        $.getJSON(OSM.NOMINATIM_URL + "reverse?lat=" + latlng_array[0] + "&lon=" + latlng_array[1] + "&format=json&viewbox=" + viewbox, function (json) {
+          endpoint.awaitingGeocode = false;
+          endpoint.hasGeocode = true;
+
+          endpoint.setLatLng(L.latLng(latlng_array), false, true);
+
+          if (!json || !json.display_name) {
+            input.addClass("is-invalid");
+            alert(I18n.t("javascripts.directions.errors.no_place", { place: endpoint.value }));
+            return;
+          }
+
+          if (endpoint.latlng.distanceTo(L.latLng(json.lat, json.lon)) > 200.0) {
```

Comment:
What was this distance check about? Was it to filter out useless reverse geocoding results that are too far away from the endpoint?

Reply:
Yes, exactly! If it is found something near (e.g. 200m) the clicked coordinates, accept it, otherwise, display coordinates.

Reply:
I don't do this in #5064 because it's a different problem, but it probably should be solved somewhere.

What's "near" probably depends on the current zoom level if the coordinates came from a click on the map.